analysis/tbcshaman/src/modules/spells/ChainHeal.tsx

```diff
@@ -0,0 +1,264 @@
+import Events, { CastEvent, EventType, HealEvent } from 'parser/core/Events';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import Combatants from 'parser/shared/modules/Combatants';
+import BoringValue from 'parser/ui/BoringValueText';
+import DonutChart from 'parser/ui/DonutChart';
+import Statistic from 'parser/ui/Statistic';
+import { STATISTIC_ORDER } from 'parser/ui/StatisticBox';
+import React from 'react';
+
+import lowRankSpells from '../../lowRankSpells';
+import * as SPELLS from '../../SPELLS';
+
+const CHAIN_HEAL_TARGET_EFFICIENCY = 0.97;
+const HEAL_WINDOW_MS = 250;
+
+interface ChainHealInfo {
+  target: {
+    id: number | undefined;
+    name: string;
+    specIcon: string;
+    specClassName: string;
+  };
+  timestamp: number;
+  castNo: number;
+  hits: number;
+}
+
+class ChainHeal extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+    combatants: Combatants,
+  };
+
+  protected abilityTracker!: AbilityTracker;
+  protected combatants!: Combatants;
+
+  buffer: Array<HealEvent | CastEvent> = [];
+  chainHealHistory: ChainHealInfo[] = [];
+  castIndex = 0;
+  chainHealTimestamp = 0;
+  maxTargets = 3;
+  suggestedTargets = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.suggestedTargets = this.maxTargets * CHAIN_HEAL_TARGET_EFFICIENCY;
+
+    const chainHealSpells: SpellInfo[] = [
+      SPELLS.CHAIN_HEAL,
+      ...lowRankSpells[SPELLS.CHAIN_HEAL],
+    ].map((spell) => ({ id: spell }));
+
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(chainHealSpells), this.chainHeal);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(chainHealSpells), this.chainHeal);
+  }
+
+  chainHeal(event: HealEvent | CastEvent) {
+    if (!this.chainHealTimestamp || event.timestamp - this.chainHealTimestamp > HEAL_WINDOW_MS) {
+      this.processBuffer();
+      this.chainHealTimestamp = event.timestamp;
+    }
+
+    this.buffer.push({
+      ...event,
+    });
+  }
+
+  processBuffer() {
+    if (this.buffer.length === 0) {
+      return;
+    }
+    this.castIndex += 1;
+    const currentCast = this.buffer.find((event) => event.type === EventType.Cast);
+    if (!currentCast) {
+      return;
+    }
+    const combatant = this.combatants.getEntity(currentCast);
+    if (!combatant) {
+      return;
+    }
+    this.chainHealHistory[this.castIndex] = {
+      target: {
+        id: currentCast.targetID,
+        name: combatant.name,
+        specIcon: combatant.player.icon,
+        specClassName: combatant.player.type,
+      },
+      timestamp: currentCast.timestamp,
+      castNo: this.castIndex,
+      hits: this.buffer.filter((event) => event.type === EventType.Heal).length,
+    };
+    this.buffer = [];
+  }
+
+  suggestions(when: When) {
+    const suggestedThreshold = this.suggestionThreshold;
+    if (isNaN(suggestedThreshold.actual)) {
```

Comment:
instead of using `isNaN` here, you should check the condition that you actually care about (which I think is `this.casts === 0`?)